src/tasks/generate-typescript-typedefs.ts

```diff
@@ -0,0 +1,79 @@
+import chalk from "chalk";
+import fs from "fs";
+import type { GenerateTypescriptTypedefsCLIOptions, JSONSchemaToTSOptions } from "../types";
+import { GenerateTypesFromJSONSchemas } from "../utils/typescript/generateTypesFromJSONSchema";
+import type { JSONSchema } from "json-schema-to-typescript";
+
+type GenerateTSTypedefs = (options: GenerateTypescriptTypedefsCLIOptions) => void;
+
+const generateTypescriptTypedefs: GenerateTSTypedefs = async ({
+  sourceFilePaths,
+  destinationFilePath = "./storyblok-component-types.d.ts",
```

Comment:
I recommend that to generate the new file, you add the space number, as is done in the pull-components command.
```sh
# To generate files like that...
./storyblok-component-types_SPACE_ID.d.ts
```

I recommend this because many users have components for several spaces and without it the script will overwrite itself every time the user runs the command.

Reply:
I can see your point, but I am not entirely sure about that 🤔 
There isn't a direct correlation between Typescript type definitions and spaces, unlike the relationship between components JSONSchemas and spaces.

Those type definitions can host multiple types coming from different sources. The `sourceFilePaths` is indeed plural - notice the "s" - so that a user can pull multiple sources of components (that could be both single components or a whole group of components coming from a single space) and have them all included in the same definitions. For this reason, I think that labeling the typedefs file with the space ID by default could be a bit deceiving.

A couple of positive side effects are also deduplication (you won't have multiple `AssetStoryblok` provided types) and ease of import (one single source of import for all the Storyblok component types).

WDYT?

Reply:
I see, it makes sense, so I think we can keep it and in the future we can improve it if necessary.